reference/functions/pgroonga-condition.md

```diff
@@ -0,0 +1,112 @@
+---
+title: pgroonga_condition function
+upper_level: ../
+---
+
+# `pgroonga_condition()` function
+
+## Summary
+
+`pgroonga_condition()`関数は`pgroonga_condition`型の値を返します。
+関数名と型名が同じですが別物です。
+`pgroonga_condition`型は`pgroonga_full_text_search_condition`型や`pgroonga_full_text_search_condition_with_scorers`型のように複雑な条件式を表現します。
+
+`pgroonga_condition()`関数は`pgroonga_condition`型の値を作るための便利関数です。
+特定の属性値のみを指定して`pgroonga_condition`型の値を作れます。
+
+`pgroonga_full_text_search_condition`型や`pgroonga_full_text_search_condition_with_scorers`型では、このような便利関数がなかったため
+必ず全ての属性値を指定して値を作る必要がありました。
+
+したがって、不要な属性値があっても、`pgroonga_full_text_search_condition`型や`pgroonga_full_text_search_condition_with_scorers`型では、
+以下のように不要な属性値には`NULL`を指定する必要がありました。
+
+```
+title &@~ ('keyword', NULL, 'index_name')::pgroonga_full_text_search_condition
+title &@~ ('keyword', ARRAY[1,1,1,5,0], NULL, 'index_name')::pgroonga_full_text_search_condition_with_scorers
+```
+
+型を指定して直接値を作る従来の方法では後方互換性を維持したまま新しい属性を作ることができませんでした。
+そのため、新しい属性を追加するたびに`pgroonga_full_text_search_condition_with_XXX`というような新しい型を追加する必要がありました。
+たとえば、`pgroogna_full_text_search_condition_with_scorers`型はそのために追加された型です。
+
+`pgroonga_full_text_search_condition`型と`pgroonga_full_text_search_condition_with_scorers`型
+の違いは`scorers`が存在するかどうかですが、`pgroonga_full_text_search_condition`型に`scorers`を追加してしまうと、
+`scorers`を使わないユーザーも新たに`NULL`を挿入して`pgroonga_full_text_search_condition`型の値を作らなければなりません。
+
+しかし、`pgroonga_condition`型の値を作るための便利関数`pgroonga_condition()`関数を導入することにより後方互換性を維持したまま
+`pgroonga_condition`型に新しい属性を追加できます。
+`pgroonga_condition()`関数が非互換を吸収してくれるからです。
+
```

Comment:
32-38

The difference between `pgroonga_full_text_search_condition` type and `pgroonga_full_text_search_condition_with_scorers` type is whether `scorers` exist or not. If `scorers` is added to `pgroonga_full_text_search_condition` type, every users are required to insert new 'NULL' to make `pgroonga_full_text_search_condition` type regardless of `scorers` usage.

However, installing `pgroonga_condition()` function to make new `pgroonga_condition` type value let a new attribution to be added while keeping backward compatibility.
It is because `pgroonga_condition()` function absorb incompatibility.